contracts/src/types/Selector.sol

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2025 RISC Zero, Inc.
+//
+// All rights reserved.
+pragma solidity ^0.8.20;
+
+using SelectorLib for Selector global;
+using SelectorsLib for Selectors global;
+
+/// @title Selector - A type for handling optional bytes4 values
+/// @notice Provides a safe way to handle optional values in Solidity
+struct Selector {
+    /// @notice The wrapped value, only valid when isSome is true
+    bytes4 value;
+    /// @notice Indicates whether the Option contains a value (true) or is None (false)
+    bool isSome;
+}
```

Comment:
Does this mean 0 is a valid selector? Curious why we need this

Reply:
Yeah, my idea was to enable testing it with the MockVerifier that typically uses 0 as selector. But tbh can be set with any selector as well, so I've dropped this type in favor of just a `bytes4`